### ISSUE TITLE
Changing UDP precision can corrupt default timestamp decoding

### ISSUE THREAD

**Issue by jleclanche** [influxdata/influxdb#8424]
### Bug report

__System info:__ InfluxDB 1.2, Debian Stretch

__Steps to reproduce:__

I started by creating a new database, importing old data into it, which had been input using influx's http api, most of them with precision=ms.
I then enabled the udp api, didn't specify a precision and started writing measurements **without including a timestamp**. Everything was okay; I then set `precision="s"`, everything was still okay but at that point I found out I needed higher precision accuracy, so I switched the udp input to `ms` (still not sending timestamps through udp).

In doing so, all new timestamps are now off by a factor of 1000. I then changed udp.precision back to an empty string, but timestamps are now still off by 1000x. In fact, if I change it to anything else than `"s"`, it's broken.

I don't understand how this behaviour makes sense. The UDP writer should know what unit to use to write to the database; the input `precision` setting should only be used for decoding timestamp values. When a timestamp value is not specified, it should set the precision of the value it generate, but when writing it to influx it should still know to write a correct timestamp.

In other words, setting the precision without specifying a timestamp should never write bad timestamps; and `precision=""` should *especially* not write bad timestamps.

**Comment by jleclanche** [comment]
Thinking through the code:

1. The parsing happens [here in service.go](https://github.com/influxdata/influxdb/blob/108c7bc54d6265c2766d6ed6f83d5cc8b3bedef7/services/udp/service.go#L214), using `ParsePointsWithPrecision` with a default `time.Now().UTC()`.
2. This jumps to [parsePoint](https://github.com/influxdata/influxdb/blob/3b700863eaa3fae65a995ce65ca12fc85c0a6995/models/points.go#L305), passing the precision argument.
3. [We call scanTime(), which sets the .ts attribute to the timestamp string](https://github.com/influxdata/influxdb/blob/3b700863eaa3fae65a995ce65ca12fc85c0a6995/models/points.go#L348)
4. We then create a point as pt, pt.ts is the timestamp string
5. [if pt.ts is empty](https://github.com/influxdata/influxdb/blob/3b700863eaa3fae65a995ce65ca12fc85c0a6995/models/points.go#L359-L361), pt.time is set to `defaultTime` and `pt.SetPrecision(precision)` is called.
6. Doing so [truncates the time to match that precision](https://github.com/influxdata/influxdb/blob/3b700863eaa3fae65a995ce65ca12fc85c0a6995/models/points.go#L1473-L1487).

**Comment by jleclanche** [comment]
okay, wtf? I dropped and recreated the database and it still has the same issue unless I use `s` precision.
